tests/lang/s02_core/s99_coroutines/t21_coroutine_value.nim

```diff
@@ -0,0 +1,25 @@
+
+# XXX: maybe coroutine values should be named "first-class coroutine symbol"?
```

Comment:
if `proc foo() = discard` is a `proc def`, then shouldn't `proc bar() {.coroutine.} = discard` be a `coroutine def`, although a better term might be `coroutine-constructor`?

Reply:
The current terminology I'm using (it changed a bit since writing the comment):
* `proc c() {.coroutine.} = discard` is a *coroutine definition*
* `c` is the name of a *coroutine*
* the `c()` in `launch c()` is a *coroutine instance construction*
* `proc() {.coroutine.}` is a *coroutine type*
* `launch` returns a *coroutine instance*
* with `let v = c`, `v` stores a *coroutine value* (*coroutine* alone probably suffices)